src/Mvc/test/Mvc.IntegrationTests/ComplexRecordIntegrationTest.cs

```diff
@@ -0,0 +1,4285 @@
+
+            // Act
+            var modelBindingResult = await parameterBinder.BindModelAsync(
+                testContext,
+                modelBinder,
+                valueProvider,
+                parameter,
+                metadata,
+                value: null);
+
+            // Assert
+            Assert.True(modelBindingResult.IsModelSet);
+
+            var model = Assert.IsType<RecordTypeWithSettableProperty1>(modelBindingResult.Model);
+            Assert.Null(model.Name);
+            Assert.Equal(0, model.Age);
+
+            Assert.Empty(modelState);
+            Assert.Equal(0, modelState.ErrorCount);
+            Assert.True(modelState.IsValid);
+        }
+
+        [Fact]
+        public async Task RecordTypeWithBoundParametersAndProperties_ValueForParameter()
+        {
+            // Arrange
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(RecordTypeWithSettableProperty1)
+            };
+
+            // Need to have a key here so that the ComplexObjectModelBinder will recurse to bind elements.
+            var testContext = ModelBindingTestHelper.GetTestContext(request =>
+            {
+                request.QueryString = new QueryString("?name=TestName");
+            });
+
+            var modelState = testContext.ModelState;
+            var metadata = GetMetadata(testContext, parameter);
+            var modelBinder = GetModelBinder(testContext, parameter, metadata);
+            var valueProvider = await CompositeValueProvider.CreateAsync(testContext);
+            var parameterBinder = ModelBindingTestHelper.GetParameterBinder(testContext);
+
+            // Act
+            var modelBindingResult = await parameterBinder.BindModelAsync(
+                testContext,
+                modelBinder,
+                valueProvider,
+                parameter,
+                metadata,
+                value: null);
+
+            // Assert
+            Assert.True(modelBindingResult.IsModelSet);
+
+            var model = Assert.IsType<RecordTypeWithSettableProperty1>(modelBindingResult.Model);
+            Assert.Equal("TestName", model.Name);
+            Assert.Equal(0, model.Age);
+
+            var entry = Assert.Single(modelState);
+            Assert.Equal("Name", entry.Key);
+            Assert.Equal(0, modelState.ErrorCount);
+            Assert.True(modelState.IsValid);
+        }
+
+        [Fact]
+        public async Task RecordTypeWithBoundParametersAndProperties_ValueForProperty()
+        {
+            // Arrange
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(RecordTypeWithSettableProperty1)
+            };
+
+            // Need to have a key here so that the ComplexObjectModelBinder will recurse to bind elements.
```

Comment:
Another copy-'n-paste error❔ This looks consistent in the last few tests though incorrect in many (all❔) cases.